milabench/cli.py

```diff
@@ -319,4 +329,34 @@ def dockerfile_template(milabench_req, include_data, use_conda, python_version):
 { 'RUN milabench prepare /bench/bench.yaml' if include_data else '' }
 
 CMD ["milabench", "run", "/bench/bench.yaml"]
-"""
\ No newline at end of file
+"""
+
+def singularitydef_template(milabench_req, include_data, use_conda, python_version):
+    return f"""\
+BootStrap: docker
+From: { 'continuumio/miniconda3' if use_conda else f'python:{python_version}-slim' }
+
+%environment
+    export MILABENCH_BASE=/base
+    export VIRTUAL_ENV=/base/venv
+    export MILABENCH_DEVREQS=/version.txt
+    export WORKDIR=/base
+
+%post
+    apt-get update && apt-get install --no-install-suggests --no-install-recommends -y git patch
+    rm -rf /var/lib/apt/lists/*
+    mkdir /bench
+    mkdir /base
+    echo '{ milabench_req }' > /version.txt
+    pip install -U pip && \
+    pip install -r /version.txt && \
+    milabench install /bench/bench.yaml && \
+    rm -rf /root/.cache/pip
+
+%startscript
+{ '    milabench prepare /bench/bench.yaml' if include_data else '' }
```

Comment:
This should be in post.  The idea is to have the dataset as part of the image if `include_data` is True